x/metadata/keeper/mocks_test.go

```diff
@@ -325,14 +332,323 @@
 
 // String returns a string of this MockAuthorization and satisfies the authz.Authorization interface.
 func (a *MockAuthorization) String() string {
-	_ = MockAuthorization{
-		Name:              "",
-		AcceptResponse:    authz.AcceptResponse{},
-		AcceptResponseErr: nil,
-		AcceptCalls:       nil,
-	}
 	return fmt.Sprintf("MockAuthorization{%q, %v, %v, %d}", a.Name, a.AcceptResponse, a.AcceptResponseErr, a.AcceptCalls)
 }
 
 // ProtoMessage satisfies the authz.Authorization interface.
 func (a *MockAuthorization) ProtoMessage() {}
+
+var _ keeper.MarkerKeeper = (*MockMarkerKeeper)(nil)
+
+// MockMarkerKeeper is a mocked keeper.MarkerKeeper.
+type MockMarkerKeeper struct {
+	IsMarkerAccountResults map[string]bool
+	IsMarkerAccountCalls   []sdk.AccAddress
+}
+
+func NewMockMarkerKeeper() *MockMarkerKeeper {
+	return &MockMarkerKeeper{
+		IsMarkerAccountResults: make(map[string]bool),
+	}
+}
+
+// WithIsMarkerAccountResults sets up the provided addrs to return true from IsMarkerAccount.
+func (k *MockMarkerKeeper) WithIsMarkerAccountResults(addrs ...sdk.AccAddress) *MockMarkerKeeper {
+	for _, addr := range addrs {
+		k.IsMarkerAccountResults[string(addr)] = true
+	}
```

Comment:
See all the other comments on why I use `string(addr)` for these keys instead of `addr.String()`. The casting is faster and the content of the result are only being used as `map` keys, which work just fine with unprintable characters.